### PR TITLE
Rewrote TestLAMMPSDATAWriter to use pytest fixture

### LINKED ISSUE #1501: Python 3 `hasattr` works differently, causes tests to fail
``` python
import MDAnalysis as mda
from MDAnalysisTests.datafiles import LAMMPScnt

# create a Universe without velocities
u = mda.Universe(LAMMPScnt)

# in python 2
In [7]: hasattr(u.atoms, 'velocities')
Out[7]: False

# in python 3
In [5]: hasattr(u.atoms, 'velocities')
NoDataError: Timestep does not contain velocities


```

This means we can't use `hasattr(u.atoms, 'velocities')` to check for the presence of velocities any more. Instead we should use a try/except block (which is arguably better Python anyway)

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #1501

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by richardjgowers** [comment]
This fixes 2 of the failures in the Python 3 build.  If I'm entirely honest, I don't know why, I rewrote the tests to start to debug them, and they fixed themselves.  There's no drop in coverage so I don't think I've broken anything by refactoring the tests

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 1]
```diff
@@ -345,82 +341,64 @@ def test_Writer_is_LAMMPS(self):
...
 
-    def test_write_non_numerical_types(self):
+    def test_write_non_numerical_types(self, tmpout):
         u = make_Universe(('types', 'masses'), trajectory=True)
 
         try:
-            u.atoms.write(self.outfile)
+            u.atoms.write(tmpout)
         except ValueError as e:
             assert_('must be convertible to integers' in e.args[0])
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: There are some old school asserts left

Comment (richardjgowers) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: Yeah I'm just trying to get py3 working

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 2]
```diff
@@ -324,6 +319,7 @@ def test_OtherWriter(self):
                             self.u.trajectory[0].positions,
                             6, err_msg="coordinate mismatch between corresponding frames")
 
+
```
Comment (tylerjereddy) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: Please remove this whitespace

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 3]
```diff
@@ -345,82 +341,64 @@ def test_Writer_is_LAMMPS(self):
...
-            except Exception:
-                return False
-            else:
-                return True
-        assert_(open_dcd(), True)
+        try:
+            with mda.coordinates.LAMMPS.DCDWriter(self.outfile, n_atoms=10):
+                pass
+        except Exception:
+            pytest.fail()
```
Comment (tylerjereddy) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: This is a bit awkward, but probably ok; see: https://stackoverflow.com/a/20275035

**Comment by tylerjereddy** [comment]
Apparently part of the Py 2/3 discrepancy is the suppression of `hasattr()` exceptions before Python 3.1: https://stackoverflow.com/a/9748889

**Pull request merged**